qhbmlib/hamiltonian_infer_utils.py

```diff
@@ -106,6 +106,7 @@ def fidelity(model: hamiltonian_model.Hamiltonian, sigma: tf.Tensor):
     A scalar `tf.Tensor` which is the fidelity between the density matrix
       represented by this QHBM and `sigma`.
   """
+  sigma = tf.cast(sigma, tf.complex64)
```

Comment:
Nice!  Since the function now accepts `sigma` with dtype beyond `tf.complex64`, can you update the docstring to reflect this?

Reply:
Sure! Working on it

Reply:
Would this work for the docstring?
```diff
  Args:
    model: Modular Hamiltonian whose corresponding thermal state is to be
      compared to `sigma`, as the left density matrix in fidelity.
-    sigma: 2-D `tf.Tensor` of dtype `complex64` representing the right
+    sigma: 2-D `tf.Tensor` of a numeric dtype representing the right
      density matrix in the fidelity calculation.
```